physionet-django/project/modelcomponents/publishedproject.py

```diff
@@ -210,7 +210,19 @@ def can_view_files(self, user):
         """
         Whether the user has access to this project and files. Extends 'is_authorized'
         """
-        return self.allow_file_downloads and self.is_authorized(user)
+        is_accessible = DataSource.objects.filter(
+            project=self,
+            data_location=DataSource.DataLocation.DIRECT).exists()
+        return self.is_authorized(user) and is_accessible
+
+    def can_create_research_environment(self, user):
+        """
+        Whether the user has access to create a research environment for this project.
+        """
+        is_accessible = DataSource.objects.filter(
+            project=self,
+            access_mechanism=DataSource.AccessMechanism.RESEARCH_ENVIRONMENT).exists()
```

Comment:
`is_accessible` sounds a bit misleading when compared to the manager's `accessible_by` method - one is about authorization, the other is about configuration.

Maybe something like `has_research_environment_source` would be more explicit?